HRInvoiceApp/HRInvoiceApp/AddCompany.xaml.cs

```diff
@@ -19,27 +20,80 @@ public partial class AddCompany : ContentPage
 
         Department department;
         KvK kvk;
-        Province province;
+        List<Province> provinces;
         User user;
         
-        public AddCompany ()
+        public AddCompany()
 		{
-			InitializeComponent ();
+			InitializeComponent();
             SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = AppViewBackButtonVisibility.Visible;
             db = App.Database.GetInstance();
 
             Task.Run(async () =>
             {
-                company = await db.Table<Company>().FirstOrDefaultAsync();
+                user = await db.Table<User>().FirstOrDefaultAsync();
+                provinces = await db.Table<Province>().ToListAsync();
+                if(provinces.Count == 0)
+                {
+                    // Error handling
+                }
+
+                if(user == null)
+                {
+                    await DisplayAlert("Alert", "Voeg eerst een gebruiker toe.", "OK");
+                    var modalPage = new SettingsPage();
+                    await Navigation.PushModalAsync(modalPage);
+                    return;
+                }
                 company = new Company();
```

Comment:
Omdat je deze hier zet, en je wil nadat je een nieuwe company hebt toegevoegd WEER 1 toevoegen, moet je eerst de view weer resetten (omdat die new company() in de constructor staat.)